catkit2/services/oceanoptics_spectrometer/oceanoptics_spectrometer.py

```diff
@@ -0,0 +1,187 @@
+        The model of the spectrometer.
+    pixels_number : int
+        number of pix in the spectrum.
+    wavelengths : NDArray[numpy.float_]
+        A NDArray with the wavelengths of the spectrometer.
+    spectra : DataStream
+        A data stream to submit spectra from the spectrometer.
+    should_be_acquiring : threading.Event
+        An event to signal whether the spectrometer should acquire a spectrum.
+    NUM_FRAMES : int
+        The number of frames to allocate for the data streams.
+
+    Methods
+    -------
+    open()
+        Open the service.
+    main()
+        The main function of the service.
+    close()
+        Close the service.
+    start_acquisition()
+        Start the acquisition.
+    take_one_spectrum()
+        Get a spectrum from the spectrometer.
+    exposure_time()
+        Set the exposure time of the spectrometer.
+    '''
+    NUM_FRAMES = 20
+
+    def __init__(self):
+        '''
+        Create a new OceanOpticsSpectrometer service.
+        '''
+        super().__init__('oceanoptics_spectrometer')
+
+        self.serial_number = None
+        self.spectrometer = None
+        self.model = None
+        self.pixels_number = None
+
+        self.wavelengths = None
+        self.spectra = None
+
+        self._exposure_time = None
+
+        self.should_be_acquiring = threading.Event()
+        self.should_be_acquiring.set()
+
+    def open(self):
+        '''
+        Open the service.
+
+        This function is called when the service is opened.
+        It initializes the spectrometer and creates the data streams and properties.
+
+        Raises
+        ------
+        RuntimeError
+            If the spectrometer cannot be found.
+        '''
+
+        # Find the spectrometer
+        self.serial_number = str(self.config['serial_number'])
+        try:
+            self.spectrometer = Spectrometer.from_serial_number(self.serial_number)
+        except SeaBreezeError:
+            raise ImportError(f'OceanOptics: Could not find spectrometer with serial number {self.serial_number}')
+
+        # exctract attributes
+        self.model = self.spectrometer.model
+        self.pixels_number = self.spectrometer.pixels
+        self.wavelengths = self.spectrometer.wavelengths()
+
+        # Define and set defaut exposure time
+        self._exposure_time = self.config.get('exposure_time', 1000)
+        self.exposure_time(self._exposure_time)
+
+        # Create datastreams
+        # Use the full sensor size here to always allocate enough shared memory.
+        self.spectra = self.make_data_stream('spectra', 'float32', [self.pixels_number], self.NUM_FRAMES)
+
+    def main(self):
+        '''
+        The main function of the service.
+
+        This function is called when the service is started.
+        '''
+        while not self.should_shut_down:
+            if self.should_be_acquiring.wait(0.05):
+                self.take_one_spectrum()
+                self.end_acquisition()
```

Comment:
It still looks to me like this will take exactly one spectrum and then end the acquisition.

I learned today that the power meter service simplifies this because it doesn't need a super fast acquisition like a camera, see here (and following lines):
https://github.com/spacetelescope/catkit2/blob/98758710dc8696a6d3d3e78aa2a1bf408bf720d5/catkit2/services/thorlabs_pm/thorlabs_pm.py#L20

What it does is that it simply defines an interval in the config entry after which it regularly takes a new measurement. I feel like this would also be the best solution for the spectrometer, as it allows you to ditch the star/end acquisition logic and just have a simple while look in `main()`.